library/rsa.c

```diff
@@ -1820,19 +1822,26 @@ int mbedtls_rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
 
     hlen = mbedtls_md_get_size( md_info );
 
-    /* Calculate the largest possible salt length. Normally this is the hash
-     * length, which is the maximum length the salt can have. If there is not
-     * enough room, use the maximum salt length that fits. The constraint is
-     * that the hash length plus the salt length plus 2 bytes must be at most
-     * the key length. This complies with FIPS 186-4 §5.5 (e) and RFC 8017
-     * (PKCS#1 v2.2) §9.1.1 step 3. */
-    min_slen = hlen - 2;
-    if( olen < hlen + min_slen + 2 )
-        return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
-    else if( olen >= hlen + hlen + 2 )
-        slen = hlen;
+    if (saltlen == MBEDTLS_RSA_SALT_LEN_ANY)
+    {
+        /* Calculate the largest possible salt length. Normally this is the hash
+        * length, which is the maximum length the salt can have. If there is not
```

Comment:
Preexisting (my fault):
```suggestion
        /* Calculate the largest possible salt length, up to the hash size.
         * Normally this is the hash length, which is the maximum salt length
        *  according to FIPS 185-4 §5.5 (e) and common practice. If there is not
```

Reply:
I will adapt the documentation as you suggested.